src/main/java/org/sagebionetworks/bridge/spring/controllers/SharedAssessmentController.java

```diff
@@ -0,0 +1,168 @@
+        return AssessmentDto.create(retValue);
+    }
+
+    @GetMapping("/v1/sharedassessments/identifier:{identifier}/revisions/{revision}")
+    public AssessmentDto getSharedAssessmentById(@PathVariable String identifier, @PathVariable String revision) {
+        // 0 is not a valid value, on purpose, will throw BadRequestException
+        int revisionInt = BridgeUtils.getIntOrDefault(revision, 0);
+        
+        Assessment retValue = service.getAssessmentById(SHARED_STUDY_ID_STRING, identifier, revisionInt);
+        return AssessmentDto.create(retValue);
+    }
+
+    @GetMapping("/v1/sharedassessments/{guid}/revisions")
+    public PagedResourceList<AssessmentDto> getSharedAssessmentRevisionsByGuid(
+            @PathVariable String guid,
+            @RequestParam(required = false) String offsetBy, 
+            @RequestParam(required = false) String pageSize,
+            @RequestParam(required = false) String includeDeleted) {
+        
+        int offsetByInt = BridgeUtils.getIntOrDefault(offsetBy, 0);
+        int pageSizeInt = BridgeUtils.getIntOrDefault(pageSize, API_DEFAULT_PAGE_SIZE);
+        boolean incDeletedBool = Boolean.valueOf(includeDeleted);
+        
+        PagedResourceList<Assessment> page = service.getAssessmentRevisionsByGuid(
+                SHARED_STUDY_ID_STRING, guid, offsetByInt, pageSizeInt, incDeletedBool);
+
+        List<AssessmentDto> dtos = page.getItems().stream()
+                .map(assessment -> AssessmentDto.create(assessment))
+                .collect(Collectors.toList());
+        
+        return new PagedResourceList<>(dtos, page.getTotal())
+                .withAllRequestParams(page.getRequestParams());
+    }
+
+    @GetMapping("/v1/sharedassessments/identifier:{identifier}/revisions")
+    public PagedResourceList<AssessmentDto> getSharedAssessmentRevisionsById(
+            @PathVariable String identifier,
+            @RequestParam(required = false) String offsetBy, 
+            @RequestParam(required = false) String pageSize,
+            @RequestParam(required = false) String includeDeleted) {
+        
+        int offsetByInt = BridgeUtils.getIntOrDefault(offsetBy, 0);
+        int pageSizeInt = BridgeUtils.getIntOrDefault(pageSize, API_DEFAULT_PAGE_SIZE);
+        boolean incDeletedBool = Boolean.valueOf(includeDeleted);
+        
+        PagedResourceList<Assessment> page = service.getAssessmentRevisionsById(
+                SHARED_STUDY_ID_STRING, identifier, offsetByInt, pageSizeInt, incDeletedBool);
+
+        List<AssessmentDto> dtos = page.getItems().stream()
+                .map(assessment -> AssessmentDto.create(assessment))
+                .collect(Collectors.toList());
+        
+        return new PagedResourceList<>(dtos, page.getTotal())
+                .withAllRequestParams(page.getRequestParams());
+    }
+
+    @PostMapping("/v1/sharedassessments/{guid}")
+    public AssessmentDto updateSharedAssessment(@PathVariable String guid) {
```

Comment:
guid isn't used anywhere.

Reply:
Updated to set the GUID on the submitted assessment (in all our APIs, the values in the URL take precedence over what is submitted in the JSON body).